src/utils/getTileProperties.ts

```diff
@@ -12,13 +12,44 @@ export const getTileProperties = (
   zoom: number
 ): TileProperties => {
   const y = row * boxHeight + tileYOffset;
-  const parsedResourceStartDate = dayjs(resourceStartDate).hour(0).minute(0);
-  const parsedResourceEndDate = dayjs(resourceEndDate).hour(23).minute(59);
+  let parsedResourceStartDate;
+  let parsedResourceEndDate;
+
+  switch (zoom) {
+    case 2: {
+      parsedResourceStartDate = dayjs(resourceStartDate);
+      parsedResourceEndDate = dayjs(resourceEndDate);
+      break;
+    }
+    default: {
+      parsedResourceStartDate = dayjs(resourceStartDate).hour(0).minute(0);
+      parsedResourceEndDate = dayjs(resourceEndDate).hour(23).minute(59);
+      break;
+    }
+  }
+
+  const rangeStartHour = startDate.hour();
+  const rangeEndHour = endDate.hour();
+  let parsedStartDate;
+  let parsedEndDate;
+
+  switch (zoom) {
+    case 2: {
+      parsedStartDate = dayjs(startDate).hour(rangeStartHour).minute(0);
+      parsedEndDate = dayjs(endDate).hour(rangeEndHour).minute(0);
+      break;
+    }
+    default: {
+      parsedStartDate = startDate;
+      parsedEndDate = endDate;
+      break;
+    }
+  }
```

Comment:
Same here? I think you can encapsulate both switch/cases in one if/else statement